File: libAndroid/com/earndero/sqlwrapper/android/Table.java
package com.earndero.sqlwrapper.android;

import java.util.ArrayList;
import java.util.List;

public class Table {
    Database db;
    final String name;
    private boolean created;
    private List<Column> columns = new ArrayList<>();
    private List<Index> indices = new ArrayList<>();
    String[] columnNames;

    Table(boolean created, String name, Database db) {
        this.created = created;
        this.name = name;
        this.db = db;
    }

    public void addColumn(String name, String type) {
        columns.add(new Column(name, type));
    }

    public void create() throws SQLException {
        if (created) throw new SQLException();
        String stmt = "create table \""+ name + "\"(";
        for (int i=0; i<columns.size(); i++) {
            if (i > 0)
                stmt += ",";
            Column col = columns.get(i);
            stmt += "\""+col.name +"\"" + " " + col.type;
        }
        stmt += ")";
        for (int i=0; i<indices.size(); i++) {
            stmt += ";";
            Index idx = indices.get(i);
            stmt += "CREATE INDEX \""+idx.mName+"\" on \"" + name +"\" (";
            for (int j=0; j<idx.Colarray.length; j++) {
                if (j > 0) stmt += ",";
                stmt += "\""+idx.Colarray[j]+"\"";
            }
            stmt += ")";
        }
        Statement statement = db.createStatement();
        statement.exec(stmt);
        columnNames = new String[columns.size()];
        for (int i=0; i<columns.size(); i++) {
            columnNames[i] = columns.get(i).name;
        }
        created = true;
    }

    public void addIndex(String name, String columns, boolean unique) {
        indices.add(new Index(name, columns, unique));
    }

    public void insert(RowValues rv) {
        StringBuilder stmt = new StringBuilder("INSERT INTO " + name + " (");
        for (int i=0; i<rv.values.size(); i++) {
            if (i>0) stmt.append(",");
            stmt.append(rv.values.get(i).colName);
        }
        stmt.append(") VALUES (");
        for (int i=0; i<rv.values.size(); i++) {
            if (i>0) stmt.append(",");
            Object v = rv.values.get(i).value;
            if (v instanceof String)
                stmt.append("\"").append((String) v).append("\"");
            else if (v instanceof Double)
                stmt.append(String.valueOf((Double) v));
            else if (v instanceof Integer)
                stmt.append(String.valueOf((Integer) v));
        }
        stmt.append(")");
        Statement statement = db.createStatement();
        statement.exec(stmt.toString());
    }
}

File: libJavaKt/com/earndero/sqlwrapper/javakt/Table.kt
package com.earndero.sqlwrapper.javakt

import kotlin.Throws
import java.lang.StringBuilder
import java.util.ArrayList

class Table internal constructor(private var created: Boolean, private val name: String, private val db: Database) {
    private val columns: MutableList<Column> = ArrayList()
    private val indices: MutableList<Index> = ArrayList()
    fun addColumn(name: String, type: String) {
        columns.add(Column(name, type))
    }

    @Throws(SQLException::class)
    fun create() {
        if (created) throw SQLException()
        var stmt = "create table \"$name\"("
        for (i in columns.indices) {
            if (i > 0) stmt += ","
            val col = columns[i]
            stmt += "\"" + col.mName + "\"" + " " + col.mType
        }
        stmt += ")"
        for (i in indices.indices) {
            stmt += ";"
            val idx = indices[i]
            stmt += "CREATE INDEX \"" + idx.mName + "\" on \"" + name + "\" ("
            for (j in idx.Colarray.indices) {
                if (j > 0) stmt += ","
                stmt += "\"" + idx.Colarray[j] + "\""
            }
            stmt += ")"
        }
        val statement = db.createStatement()
        statement!!.exec(stmt)
        created = true
    }

    fun addIndex(name: String, columns: String, unique: Boolean) {
        indices.add(Index(name, columns, unique))
    }

    fun insert(rv: RowValues) {
        val stmt = StringBuilder("INSERT INTO $name (")
        for (i in rv.values.indices) {
            if (i > 0) stmt.append(",")
            stmt.append(rv.values[i]!!.colName)
        }
        stmt.append(") VALUES (")
        for (i in rv.values.indices) {
            if (i > 0) stmt.append(",")
            val v = rv.values[i]!!.value
            if (v is String)
                stmt.append("\"").append(v).append("\"")
            else if (v is Double)
                stmt.append((v as Double).toString())
            else if (v is Int)
                stmt.append((v as Int).toString())
        }
        stmt.append(")")
        val statement = db.createStatement()
        statement!!.exec(stmt.toString())
    }
}